Build failure due to faulty attempt to install libsemigroups

When building semigroups-5.5.3 with the bundled libsemigroups 2.7.4, `make` fails:

```
20:16 a4:../gap-semigroups/semigroups-5.5.3 $ ./configure --prefix=/usr --with-gaproot=/usr/lib64/gap
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking whether g++ supports C++14 features by default... yes
checking for GAP root directory... /usr/lib64/gap
checking for GAP architecture... x86_64-suse-linux-gnu-default64-kv9
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for a sed that does not truncate output... /usr/bin/sed
checking how to run the C preprocessor... gcc -E
checking for egrep -e... /usr/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for pthread_create in -lpthread... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for gawk... gawk
configure: using included libsemigroups...
checking libsemigroups version... 2.7.4
checking whether to enable debug mode... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking whether to enable HPCombi... yes
checking the required version of HPCombi... 0.0.6
checking the version of HPCombi that's present... 0.0.6
checking whether C++ compiler accepts -mavx... yes
checking whether C++ compiler accepts -flax-vector-conversions... yes
checking for x86intrin.h... yes
checking for _mm_blendv_epi8... yes
checking for _mm_cmpeq_epi8... yes
checking for _mm_cmpestri... yes
checking for _mm_cmpestrm... yes
checking for _mm_cmplt_epi8... yes
checking for _mm_extract_epi64... yes
checking for _mm_max_epi8... yes
checking for _mm_max_epu8... yes
checking for _mm_min_epi8... yes
checking for _mm_min_epu8... yes
checking for _mm_movemask_epi8... yes
checking for _mm_popcnt_u32... yes
checking for _mm_set_epi64x... yes
checking for _mm_shuffle_epi8... yes
checking for _mm_slli_epi32... yes
checking for HPCOMBI_CONSTEXPR_FUN_ARGS... yes
configure: creating ./config.status
config.status: creating GNUmakefile
config.status: creating gen/pkgconfig.h
config.status: executing src/semigroups-config.hpp commands
config.status: creating src/semigroups-config.hpp - prefix SEMIGROUPS for gen/pkgconfig.h defines
=== configuring in libsemigroups (/home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=${abs_top_builddir}/../bin/'  '--with-gaproot=/usr/lib64/gap' --cache-file=/dev/null --srcdir=.
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether sleep supports fractional seconds... yes
checking filesystem timestamp resolution... 0.01
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking xargs -n works... yes
checking how to create a pax tar archive... gnutar
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... gcc3
checking whether g++ supports C++14 features by default... yes
checking for a sed that does not truncate output... /usr/bin/sed
checking how to run the C preprocessor... gcc -E
checking for egrep -e... /usr/bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for pthread_create in -lpthread... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether ln -s works... yes
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/x86_64-suse-linux/bin/ld
checking if the linker (/usr/x86_64-suse-linux/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/x86_64-suse-linux/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ranlib... ranlib
checking for archiver @FILE support... @
checking for strip... strip
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/x86_64-suse-linux/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/x86_64-suse-linux/bin/ld -m elf_x86_64
checking if the linker (/usr/x86_64-suse-linux/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/x86_64-suse-linux/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/x86_64-suse-linux/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for limits.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for sys/time.h... yes
checking for unistd.h... (cached) yes
checking for pthread.h... yes
checking for _Bool... yes
checking for stdbool.h that conforms to C99 or later... yes
checking for inline... inline
checking for int64_t... yes
checking for size_t... yes
checking for uint64_t... yes
checking for ptrdiff_t... yes
checking size of void *... 8
checking for GNU libc compatible malloc... yes
checking for gettimeofday... yes
checking for memset... yes
checking for pow... no
checking for sqrt... no
checking whether to enable debug mode... no
checking whether to enable verbose mode... no
checking whether enable compiler warnings... no
checking whether to enable stats mode... yes
checking whether to enable backward mode... yes
checking for execinfo.h... yes
checking whether to build with code coverage support... no
checking whether to enable HPCombi... yes
checking for ./extern/HPCombi/VERSION... yes
checking the required version of HPCombi... 0.0.6
checking the version of HPCombi that's present... 0.0.6
checking whether C++ compiler accepts -mavx... yes
checking whether C++ compiler accepts -flax-vector-conversions... yes
checking for x86intrin.h... yes
checking for _mm_blendv_epi8... yes
checking for _mm_cmpeq_epi8... yes
checking for _mm_cmpestri... yes
checking for _mm_cmpestrm... yes
checking for _mm_cmplt_epi8... yes
checking for _mm_extract_epi64... yes
checking for _mm_max_epi8... yes
checking for _mm_max_epu8... yes
checking for _mm_min_epi8... yes
checking for _mm_min_epu8... yes
checking for _mm_movemask_epi8... yes
checking for _mm_popcnt_u32... yes
checking for _mm_set_epi64x... yes
checking for _mm_shuffle_epi8... yes
checking for _mm_slli_epi32... yes
checking for HPCOMBI_CONSTEXPR_FUN_ARGS... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking whether to enable fmt... no
checking whether to enable eigen... yes
checking whether to use external eigen... no
checking for __builtin_popcountl... yes
checking for __builtin_clzll... yes
checking whether to try and use __builtin_popcntl... yes
checking whether to try and use __builtin_clzlll... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libsemigroups.pc
config.status: creating config/config.h
config.status: executing depfiles commands
config.status: executing include/libsemigroups/config.hpp commands
config.status: creating include/libsemigroups/config.hpp - prefix LIBSEMIGROUPS for config/config.h defines
config.status: executing libtool commands
20:22 a4:../gap-semigroups/semigroups-5.5.3 $ make -j32 V=0
make -C libsemigroups install
./config.status gen/pkgconfig.h
make[1]: Entering directory '/home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups'
  CXX      src/libsemigroups_la-bmat8.lo
  CXX      src/libsemigroups_la-bipart.lo
  CXX      src/libsemigroups_la-cong-intf.lo
  CXX      src/libsemigroups_la-cong-pair.lo
  CXX      src/libsemigroups_la-cong.lo
  CXX      src/libsemigroups_la-constants.lo
  CXX      src/libsemigroups_la-exception.lo
  CXX      src/libsemigroups_la-forest.lo
  CXX      src/libsemigroups_la-coset.lo
  CXX      src/libsemigroups_la-felsch-tree.lo
  CXX      src/libsemigroups_la-fpsemi-examples.lo
  CXX      src/libsemigroups_la-fpsemi-intf.lo
  CXX      src/libsemigroups_la-freeband.lo
  CXX      src/libsemigroups_la-fpsemi.lo
  CXX      src/libsemigroups_la-froidure-pin-base.lo
  CXX      src/libsemigroups_la-knuth-bendix.lo
  CXX      src/libsemigroups_la-kambites.lo
  CXX      src/libsemigroups_la-node-manager.lo
  CXX      src/libsemigroups_la-make-present.lo
  CXX      src/libsemigroups_la-obvinf.lo
  CXX      src/libsemigroups_la-pbr.lo
  CXX      src/libsemigroups_la-race.lo
  CXX      src/libsemigroups_la-report.lo
  CXX      src/libsemigroups_la-runner.lo
  CXX      src/libsemigroups_la-string.lo
  CXX      src/libsemigroups_la-siso.lo
  CXX      src/libsemigroups_la-stephen.lo
  CXX      src/libsemigroups_la-todd-coxeter.lo
  CXX      src/libsemigroups_la-timer.lo
  CXX      src/libsemigroups_la-ukkonen.lo
  CXX      src/libsemigroups_la-wilo.lo
config.status: creating gen/pkgconfig.h
config.status: gen/pkgconfig.h is unchanged
echo > gen/pkgconfig.h.stamp
  CXX      src/libsemigroups_la-wislo.lo
  CXX      src/libsemigroups_la-word.lo
In file included from /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:26,
                 from /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/cong-intf.hpp:31,
                 from /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/cong.hpp:28,
                 from src/cong.cpp:22:
In member function '_Tp std::atomic<_Tp>::load(std::memory_order) const [with _Tp = libsemigroups::Runner::state]',
    inlined from 'std::atomic<_Tp>::operator _Tp() const [with _Tp = libsemigroups::Runner::state]' at /usr/include/c++/15/atomic:253:2 ,
    inlined from 'libsemigroups::Runner::state libsemigroups::Runner::get_state() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:552:14,
    inlined from 'bool libsemigroups::Runner::started() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:397:23,
    inlined from 'bool libsemigroups::Runner::finished() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:374:18,
    inlined from 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)' at src/cong.cpp:89:37:
/usr/include/c++/15/atomic:308:22: warning: 'unsigned int __atomic_load_4(const volatile void*, int)' writing 4 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
  308 |         __atomic_load(std::__addressof(_M_i), __ptr, int(__m));
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In constructor 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)':
cc1plus: note: destination object is likely at address zero
In member function '_Tp std::atomic<_Tp>::load(std::memory_order) const [with _Tp = libsemigroups::Runner::state]',
    inlined from 'std::atomic<_Tp>::operator _Tp() const [with _Tp = libsemigroups::Runner::state]' at /usr/include/c++/15/atomic:253:2 ,
    inlined from 'libsemigroups::Runner::state libsemigroups::Runner::get_state() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:552:14,
    inlined from 'bool libsemigroups::Runner::started() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:397:23,
    inlined from 'bool libsemigroups::Runner::finished() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:374:18,
    inlined from 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)' at src/cong.cpp:107:37:
/usr/include/c++/15/atomic:308:22: warning: 'unsigned int __atomic_load_4(const volatile void*, int)' writing 4 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
  308 |         __atomic_load(std::__addressof(_M_i), __ptr, int(__m));
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In constructor 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)':
cc1plus: note: destination object is likely at address zero
In member function '_Tp std::atomic<_Tp>::load(std::memory_order) const [with _Tp = libsemigroups::Runner::state]',
    inlined from 'std::atomic<_Tp>::operator _Tp() const [with _Tp = libsemigroups::Runner::state]' at /usr/include/c++/15/atomic:253:2 ,
    inlined from 'libsemigroups::Runner::state libsemigroups::Runner::get_state() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:552:14,
    inlined from 'bool libsemigroups::Runner::dead() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:454:23,
    inlined from 'bool libsemigroups::Runner::finished() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:374:29,
    inlined from 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)' at src/cong.cpp:107:37:
/usr/include/c++/15/atomic:308:22: warning: 'unsigned int __atomic_load_4(const volatile void*, int)' writing 4 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
  308 |         __atomic_load(std::__addressof(_M_i), __ptr, int(__m));
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In constructor 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)':
cc1plus: note: destination object is likely at address zero
In member function '_Tp std::atomic<_Tp>::load(std::memory_order) const [with _Tp = libsemigroups::Runner::state]',
    inlined from 'std::atomic<_Tp>::operator _Tp() const [with _Tp = libsemigroups::Runner::state]' at /usr/include/c++/15/atomic:253:2 ,
    inlined from 'libsemigroups::Runner::state libsemigroups::Runner::get_state() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:552:14,
    inlined from 'bool libsemigroups::Runner::dead() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:454:23,
    inlined from 'bool libsemigroups::Runner::finished() const' at /home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups/include/libsemigroups/runner.hpp:374:29,
    inlined from 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)' at src/cong.cpp:89:37:
/usr/include/c++/15/atomic:308:22: warning: 'unsigned int __atomic_load_4(const volatile void*, int)' writing 4 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
  308 |         __atomic_load(std::__addressof(_M_i), __ptr, int(__m));
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In constructor 'libsemigroups::Congruence::Congruence(libsemigroups::congruence_kind, libsemigroups::FpSemigroup&)':
cc1plus: note: destination object is likely at address zero
  CXXLD    libsemigroups.la
make[2]: Entering directory '/home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups'
 /usr/bin/mkdir -p '/usr/lib'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/backward-cpp'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Cholesky'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/CholmodSupport'
 /bin/sh ./libtool   --mode=install /usr/bin/install -c   libsemigroups.la '/usr/lib'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/AltiVec'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/AVX'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/AVX512'
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/CUDA'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/Default'
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/NEON'
make[2]: *** [Makefile:3493: install-eigenincludeHEADERS] Error 1
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/SSE'
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [Makefile:3514: install-eigensrccholeskyincludeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
 /usr/bin/install -c -m 644 extern/backward-cpp/backward.hpp '/usr/include/libsemigroups/backward-cpp'
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/arch/ZVector'
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
make[2]: *** [Makefile:3535: install-eigensrccholmodsupportincludeHEADERS] Error 1
make[2]: *** [Makefile:3556: install-eigensrccorearchaltivecincludeHEADERS] Error 1
make[2]: *** [Makefile:3577: install-eigensrccorearchavx512includeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core/functors'
make[2]: *** [Makefile:3598: install-eigensrccorearchavxincludeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
make[2]: *** [Makefile:3619: install-eigensrccorearchcudaincludeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
make[2]: *** [Makefile:3640: install-eigensrccorearchdefaultincludeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
 /usr/bin/mkdir -p '/usr/include/libsemigroups/Eigen/src/Core'
install: cannot remove '/usr/include/libsemigroups/backward-cpp/backward.hpp': Permission denied
make[2]: *** [Makefile:3682: install-eigensrccorearchsseincludeHEADERS] Error 1
make[2]: *** [Makefile:3703: install-eigensrccorearchzvectorincludeHEADERS] Error 1
make[2]: *** [Makefile:3661: install-eigensrccorearchneonincludeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
make[2]: *** [Makefile:3472: install-backwardcppincludeHEADERS] Error 1
make[2]: *** [Makefile:3724: install-eigensrccorefunctorsincludeHEADERS] Error 1
mkdir: cannot create directory ‘/usr/include/libsemigroups/Eigen’: Permission denied
make[2]: *** [Makefile:3745: install-eigensrccoreincludeHEADERS] Error 1
libtool: install: /usr/bin/install -c .libs/libsemigroups.so.2.0.0 /usr/lib/libsemigroups.so.2.0.0
install: cannot create regular file '/usr/lib/libsemigroups.so.2.0.0': Permission denied
make[2]: *** [Makefile:2410: install-libLTLIBRARIES] Error 1
make[2]: Leaving directory '/home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups'
make[1]: *** [Makefile:4664: install-am] Error 2
make[1]: Leaving directory '/home/ej/obs_nosave/zu/science/gap-semigroups/semigroups-5.5.3/libsemigroups'
make: *** [GNUmakefile:86: bin/include/libsemigroups/libsemigroups.hpp] Error 2
```

--- a/GNUmakefile.in
+++ b/GNUmakefile.in
@@ -1,6 +1,11 @@
 #
 # Makefile rules for the semigroups package
 #
+abs_top_builddir = @abs_top_builddir@
+top_builddir = @top_builddir@
+abs_builddir = @abs_builddir@
+builddir = @builddir@
+
 KEXT_NAME = semigroups
 
 KEXT_CXXFLAGS = @LIBSEMIGROUPS_CFLAGS@ -std=gnu++17 -O3
@@ -88,17 +93,19 @@ $(KEXT_OBJS): bin/include/libsemigroups/libsemigroups.hpp
 
 # the following is only run if BUILT_SOURCES is wound up
 bin/include/libsemigroups/libsemigroups.hpp:
-	$(MAKE) -C libsemigroups install
-	# Cygwin only looks for DLLs in the same directory as the executable
-	# resides in. The following achieves that assuming that the GAP
-	# being used was self-compiled by the user. This supports
-	# both older GAPs with libtool (first) and GAP without libtool (second)
-        # Additionally, we rebase the dll with respect to the database to
-        # avoid fork conflicts
+	$(MAKE) -C libsemigroups install prefix='${abs_builddir}/bin' includedir='${abs_builddir}/bin/include' libdir='${abs_builddir}/bin/lib'
+	#
+	# The Win32 dynamic loader only looks for DLLs in the same directory as
+	# the executable. As a consequence, libtool puts DLLs into
+	# "${libdir}/../bin" (for both Cygwin and MinGW), and this path
+	# extension is not configurable. This explains why we test for bin/bin/
+	# rather than bin/lib.
+	#
+	# Additionally, we rebase the dll with respect to the database to
+	# avoid fork conflicts
 	if test -f bin/bin/cygsemigroups*.dll ; then rebase -s bin/bin/cygsemigroups*.dll ; fi
 	if test -f bin/bin/cygsemigroups*.dll ; then if test -d $(GAPPATH)/.libs; then cp bin/bin/cygsemigroups*.dll $(GAPPATH)/.libs/ ; fi ; fi
 	if test -f bin/bin/cygsemigroups*.dll ; then cp bin/bin/cygsemigroups*.dll $(GAPPATH)/ ; fi
-
 endif
 
 clean:

--- a/configure.ac
+++ b/configure.ac
@@ -16,10 +16,6 @@ AC_CONFIG_MACRO_DIR([m4])
 
 AX_PREFIX_CONFIG_H([src/semigroups-config.hpp],[semigroups],[gen/pkgconfig.h])
 
-dnl ## abs_top_builddir seems to hold the top build dir for the subpackage
-dnl ## libsemigroups which is why this contains ../
-AC_PREFIX_DEFAULT('${abs_top_builddir}/../bin/')
-
 dnl ##
 dnl ## Set the language
 dnl ##

